src/Aspire.Hosting/ProjectResourceBuilderExtensions.cs

```diff
@@ -557,6 +556,20 @@ public static IResourceBuilder<ProjectResource> DisableForwardedHeaders(this IRe
         return builder;
     }
 
+    /// <summary>
+    /// Set a filter that determines if environment variables are injected for a given endpoint.
+    /// By default, all endpoints are included (if this method is not called).
+    /// </summary>
+    /// <param name="builder">The project resource builder.</param>
+    /// <param name="filter">The filter callback that returns true if and only if the endpoint should be included.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<ProjectResource> WithEndpointsInEnvironment(
+        this IResourceBuilder<ProjectResource> builder, Func<EndpointAnnotation, bool> filter)
+    {
+        builder.Resource.AddEnvironmentInjectionFilter(filter);
+        return builder;
+    }
```

Comment:
If we had add and no remove, do we need a way to reset?

Reply:
That feels like a really edge scenario.

The scenario for passing multiple filters is to have two independent pieces of code say "I want to exclude this and that endpoint from env vars". It sort of makes sense, because different pieces of code may know about a different subset of endpoints, so they're not really stepping on each others' feet.

A reset would be saying: "ignore all the exclusions that other callers told you about, and include everything in the env vars". Is this really worth having a separate public method for this? If we did have it, I'd call it `WithAllEndpointsInEnvironment()`, and not take a filter at all. But personally I'd leave it out.

Reply:
If people want to make this something that is a bit more dynamic they can always add their own filter which then has its own rules mechanism which they can mutate. But I am with @davidebbo I don't think we need it.

Reply:
... having said that, we could just make this an annotation and loop over it (a well trodden patttern in Aspire).